docs/develop/java/versioning.mdx

```diff
@@ -113,43 +113,75 @@ in none of them, then they have to share the Id.
 
-To use Worker Versioning in Java, you need to do the following:
+### Assign a Build ID to your Worker and opt into versioning
 
-1. Determine and assign a Build ID to your built Worker code, and opt in to versioning.
-2. Tell the Task Queue your Worker is listening on about that Build ID, and whether its compatible with an existing Build ID.
+You should understand assignment rules before completing this step.
+See the [Worker Versioning Pre-release README.md](https://github.com/temporalio/temporal/blob/main/docs/worker-versioning.md) for more information.
 
-### Assign a Build ID to your Worker
-
-Let's say you've chosen `deadbeef` as your Build ID, which might be a short git commit hash (a reasonable choice as Build ID).
-To assign it in your Worker code, assign the following Worker Options:
+To enable worker versioning for your worker, assign the build ID--perhaps from an environment variable--and turn it on.
 
 ```java
 // ...
 WorkerOptions workerOptions = WorkerOptions.newBuilder()
-    .setBuildId("deadbeef")
+    .setBuildId(buildId)
     .setUseBuildIdForVersioning(true)
     // ...
     .build();
 Worker w = workerFactory.newWorker("your_task_queue_name", workerOptions);
 // ...
 ```
 
-That's all you need to do in your Worker code.
-Importantly, if you start this Worker, it won't receive any tasks.
-That's because you need to tell the Task Queue about your Worker's Build ID first.
+:::warning
+
+Importantly, when you start this Worker, it won't receive any tasks until you set up assignment rules.
+
+:::
+
+### Specify versions for Activities, Child Workflows, and Continue-as-New
 
-### Tell the Task Queue about your Worker's Build ID
+:::caution
+
+Java support for this feature is under construction!
+
+:::
+
+By default, Activities, Child Workflows, and Continue-as-New Workflows are run on the build of the workflow that created them if they are also configured to run on the same Task Queue.
+When configured to run on a separate task queue, they will default to using the current assignment rules.
+
+If you want to override this behavior, you can specify your intent via the `setVersioningIntent` method on the `ActivityOptions`, `ChildWorkflowOptions`, or `ContinueAsNewOptions` objects.
+
+For example, if you want an Activity to use the latest assignment rules rather than inheriting from its parent:
+
+```java
+// ...
+private final MyActivity activity =
+    Workflow.newActivityStub(
+        MyActivity.class,
+        ActivityOptions.newBuilder()
+          .setScheduleToCloseTimeout(Duration.ofSeconds(10))
+          .setVersioningIntent(VersioningIntent.VERSIONING_INTENT_USE_ASSIGNMENT_RULES)
+          // ...other options
+          .build()
+    );
+// ...
+```
+
+### Tell the Task Queue about your Worker's Build ID (Deprecated)
+
+:::caution
+
+This section is for a version of Worker Versioning that is deprecated and will go away at some point. Please redirect your attention to [Worker Versioning](/workers#worker-versioning).
```

Comment:
"for a version of Worker Versioning" could be confusing. I suggest rephrasing along the lines of "for a previous implementation of Worker Versioning" to avoid that.